core/src/execution/datafusion/expressions/negative.rs

```diff
@@ -0,0 +1,408 @@
+                                .unwrap();
+                            for i in 0..int_array.len() {
+                                if int_array.value(i) == i8::MIN || int_array.value(i) == i8::MAX {
+                                    return Err(CometError::ArithmeticOverflow {
+                                        from_type: "integer".to_string(),
+                                    }
+                                    .into());
+                                }
+                            }
+                        }
+                        DataType::Int16 => {
+                            let int_array = array
+                                .as_any()
+                                .downcast_ref::<arrow::array::Int16Array>()
+                                .unwrap();
+                            for i in 0..int_array.len() {
+                                if int_array.value(i) == i16::MIN || int_array.value(i) == i16::MAX
+                                {
+                                    return Err(CometError::ArithmeticOverflow {
+                                        from_type: "integer".to_string(),
+                                    }
+                                    .into());
+                                }
+                            }
+                        }
+                        DataType::Int32 => {
+                            let int_array = array
+                                .as_any()
+                                .downcast_ref::<arrow::array::Int32Array>()
+                                .unwrap();
+                            for i in 0..int_array.len() {
+                                if int_array.value(i) == i32::MIN || int_array.value(i) == i32::MAX
+                                {
+                                    return Err(CometError::ArithmeticOverflow {
+                                        from_type: "integer".to_string(),
+                                    }
+                                    .into());
+                                }
+                            }
+                        }
+                        DataType::Int64 => {
+                            let int_array = array
+                                .as_any()
+                                .downcast_ref::<arrow::array::Int64Array>()
+                                .unwrap();
+                            for i in 0..int_array.len() {
+                                if int_array.value(i) == i64::MIN || int_array.value(i) == i64::MAX
+                                {
+                                    return Err(CometError::ArithmeticOverflow {
+                                        from_type: "integer".to_string(),
+                                    }
+                                    .into());
+                                }
+                            }
+                        }
+                        DataType::Float16 => {
```

Comment:
I don't think we would ever see a `Float16` because Java doesn't have that type?